weaver/cli.py

```diff
@@ -329,7 +342,7 @@ class WeaverClient(object):
     auth = None  # type: AuthHandler
 
     def __init__(self, url=None, auth=None):
-        # type: (Optional[str], Optional[AuthHandler]) -> None
+        # type: (Optional[str], Optional[AuthBase]) -> None
```

Comment:
Why replace these? `AuthBase` only has this defined: 
```python
class AuthBase:
    def __call__(self, r: models.PreparedRequest) -> models.PreparedRequest: ...
```
This is very lacking in terms of abstract class, which is why `AuthHandler` with more metadata was defined.

Note also that `AnyRequest` is used, so `requests.models.PreparedRequest` cannot be assumed.

Reply:
> Why replace these? AuthBase only has this defined

Is the issue that `AuthBase` is an abstract class or would you prefer that I make `SessionAuthHandler` a subclass of `AuthHandler`?

> Note also that AnyRequest is used, so requests.models.PreparedRequest cannot be assumed.

Is there a reason why it's acceptable for the MagpieAuth class but not here? https://github.com/Ouranosinc/requests-magpie/blob/0.2.0/requests_magpie.py#L56

Reply:
Yes, making it a subclass of `AuthHandler` is preferable. 
The only reason it is acceptable for `MagpieAuth` is because it is an external component, so Weaver cannot push (explicitly) its definitions onto it.

Reply:
No. The idea is not to push it to an external component harder to discover. I just want to make sure this component does not reimplement something already supported by other means, which is my impression since the Cookie handler was created to pass the cookie by value for a similar use case as yours.